Metadata service integration

KennyZeroTech opened the pull request:
Adds the function `compareNftsToMetadataService` that uses the metadata-service to compare against a given array of Nfts

This can be tested by running the normal test code without specifying an "original" file

KennyZeroTech committed:
Adding the ability to ask the metadata service to compare nft data

KennyZeroTech committed:
Updating sample env

KennyZeroTech committed:
Removing unneeded template from AjvError

KennyZeroTech committed:
updating lock file

JamesEarle commented on src/comparers/compareNFTs.ts:
```diff
@@ -11,7 +11,10 @@ import {
   NftDiff,
   NftFileData,
```
Bit of a nitpick, but there is inconsistency with the use of NFTs, Nfts, and nfts.

JamesEarle commented on src/comparers/compareNFTs.ts:
```diff
@@ -204,3 +207,40 @@ export const compareNftFileToDatabase = async (
+  modifiedNfts: NftData[],
+  logger: Logger
+): Promise<NftBatchDiff> => {
+  const nftDataForDatabase: { domainIds: string[] } = { domainIds: [] };
```
Consider adding some typing here

KennyZeroTech replied:
Fixed

Polkm105 committed:
Removing ipfs and database stuff as it's replaced by metadata service

Polkm105 committed:
updating yarn packages

Polkm105 committed:
Removing more deprecated database access code

Polkm105 committed:
Updating env sample

RyanCross commented on src/comparers/compareNFTs.ts:
```diff
@@ -178,29 +176,39 @@ export const compareNftFiles = (
 
-export const compareNftGroupToDatabase = async (
-  modifiedNfts: NftData[],
+interface PostReturn {
```
This could have a more descriptive name

KennyZeroTech replied:
Fixed and added to types

RyanCross commented on src/comparers/compareNFTs.ts:
```diff
@@ -178,29 +176,39 @@ export const compareNftFiles = (
-  const diff = compareNftGroups(databaseNfts, modifiedNfts, logger);
-  return diff;
-};
+  const serviceData: PostReturn = (
```
Should there be some retry or error handling logic around this request?

KennyZeroTech replied:
Added error handling for if response failed

RyanCross commented:
Please address those circleci build errors before merging.

Polkm105 committed:
Addressing PR comments

- Adding retry and error handling to axios post request
- Moving PostReturn type to the types and giving it a more descriptive name of MetadataServiceResponse

Polkm105 committed:
Trying to fix circleci

Polkm105 committed:
Still trying to get circleci to work

Polkm105 committed:
still trying to figure out circleci

Polkm105 committed:
circle-ci still not working

Polkm105 committed:
changing circleci node version

Polkm105 committed:
Resetting node version in circleci

Polkm105 committed:
windows case-sensitivity can take a hike
